app/src/androidTest/java/com/github/epfl/meili/home/GoogleSignInActivityTest.kt

```diff
@@ -101,29 +113,13 @@ class GoogleSignInActivityTest {
         Intents.init()
         onView(withId(R.id.signInButton)).check(matches(isClickable())).perform(click())
 
-        val mGoogleSignInClient = GoogleSignIn.getClient(
-                MainApplication.applicationContext(),
-                getGSO()
-        )
         val resultData = Intent()
         resultData.putExtra("name", MOCK_NAME)
         val result = Instrumentation.ActivityResult(9001, resultData)
 
-        intending(IntentMatchers.filterEquals(mGoogleSignInClient.signInIntent)).respondWith(result)
-        Intents.release()
-    }
+        val mockIntent = mockService.signInIntent()
 
-    @Test
-    fun firebaseAuthWithGoogleTest() {
-        var fake_id = "1234"
-        onView(withId(R.id.signInButton)).check(matches(isClickable())).perform(click())
-
-        testRule!!.scenario.onActivity { activity ->
-            AuthenticationService.firebaseAuthWithGoogle(activity!!, fake_id)
-
-            assert(!AuthenticationService.isLoggedIn.value!!)
-            assert(AuthenticationService.name == null)
-            assert(AuthenticationService.email == null)
-        }
+        intending(IntentMatchers.filterEquals(mockIntent)).respondWith(result)
+        Intents.release()
```

Comment:
I suggest putting Intents.init() in a separate function with the @Before tag and Intents.release() in one with the @After tag since intents are being initialized and released for multiple tests.